Objects/memoryobject.c

```diff
@@ -3315,34 +3360,36 @@ memoryiter_dealloc(PyObject *self)
 static int
 memoryiter_traverse(PyObject *self, visitproc visit, void *arg)
 {
-    memoryiterobject *it = (memoryiterobject *)self;
+    memoryiterobject *it = _PyMemoryViewIter_CAST(self);
     Py_VISIT(it->it_seq);
     return 0;
 }
 
+static inline PyObject *
+memoryiter_iter_nth(PyMemoryViewObject *seq, Py_ssize_t nth, const char *fmt)
+{
+    CHECK_RELEASED(seq);
+    const Py_buffer *view = &seq->view;
+    const char *head = (const char *)seq->view.buf;
+    const char *ptr = head + view->strides[0] * nth;
+    ptr = ADJUST_PTR(ptr, view->suboffsets, 0);
+    if (ptr == NULL) {
```

Comment:
Can `ADJUST_PTR` even fail? If not, remove this--otherwise, it certainly doesn't raise an error, so you need to add one to prevent a `SystemError`.

Reply:
(well, the previous code was doing it and I didn't want to touch it but anyway I think I can)